proposals/lip-0043.md

```diff
@@ -330,35 +426,58 @@ mainchainRegParams = {
-
-    ```java
-    registrationSignatureMessage = {  
+```python
+def verify(trs: Transaction) -> None:
+    # The ownChainID property has to be an integer less than or equal to MAX_UINT32.
+    if trs.params.ownChainID > MAX_UINT32:
+        raise Exception("Invalid ownChainID property.")
+
+    # The ownName property has to contain only characters from the set [a-z0-9!@$&_.].
+    if not re.match(r"^[a-z0-9!@$&_.]+$", trs.params.ownName):
+        raise Exception("Invalid ownName property. It should contain only characters from the set [a-z0-9!@$&_.].")
+    # The name property has to be at most MAX_LENGTH_NAME characters long.
+    if len(trs.params.ownName) > MAX_LENGTH_NAME:
+        raise Exception(f"Invalid ownName property. It should be at most {MAX_LENGTH_NAME} characters long.")
+
+    # mainchainValidators must have exactly MAINCHAIN_NUMBER_ACTIVE_VALIDATORS elements.
+    if len(trs.params.mainchainValidators) != MAINCHAIN_NUMBER_ACTIVE_VALIDATORS:
+        raise Exception(f"Invalid mainchainValidators property. It must have exactly {MAINCHAIN_NUMBER_ACTIVE_VALIDATORS} elements.")
+    
+    validatorKeys = [validator["blsKey"] for validator in trs.params.initValidators]
+    # All validator keys must be valid BLS keys.
+    for key in validatorKeys:
+        if len(key) != PUBLIC_KEY_BLS_LENGTH:
+            raise Exception("Invalid BLS key length.")
+    # All validator keys must be distinct.
+    if len(validatorKeys) != len(set(validatorKeys)):
+        raise Exception("Duplicate BLS keys.")
+    
+    # Validator keys must be in lexicographic order.
+    if not all(validatorKeys[i] < validatorKeys[i + 1] for i in range(len(validatorKeys) - 1)):
+        raise Exception("Validator keys are not in lexicographic order.")
+    
+    for validator in trs.params.initValidators:
+        # The bftWeight property of each element is equal to 1.
+        if validator["bftWeight"] != 1:
+            raise Exception("Invalid bftWeight property.")
+    
+    # Check signature property.
+    height = height of the block in which trs was included
+    validators = BFT.getBFTParameters(b.header.height).validators
+    certificateThreshold = BFT.getBFTParameters(b.header.height).certificateThreshold
+    networkID = networkID of the chain
+    # Sort validators by BLS key.
+    validators.sort(key=lambda validator: validatorsModule.getValidatorAccount(validator.address).blsKey)
+    blsKeys = [validatorsModule.getValidatorAccount(validator.address).blsKey for validator in validators]
+    bftWeights = [validator.bftWeight for validator in validators]
+    
+    registrationSignatureMessageSchema = {  
         "type": "object",
         "required": ["ownChainID", "ownName", "mainchainValidators"],
         "properties": {
             "ownChainID": {
-                "dataType": "uint32",
+                "dataType": "bytes",
+                "length": CHAIN_ID_LENGTH,
```

Comment:
Reminder: All length constants `CHAIN_ID_LENGTH`, `BLS_PUBLIC_KEY_LENGTH`, `HASH_LENGTH`, `ADDRESS_LENGTH`, `BLS_SIGNATURE_LENGTH` are not defined.

Also, `BLS_PUBLIC_KEY_LENGTH` and `PUBLIC_KEY_BLS_LENGTH` are used interchangeably. 

Last, `EMPTY_HASH`, `EMPTY_BYTES` are not defined either.

Reply:
thanks, fixed the BLS constant. Other constants have been removed and added only to LIP45